pkg/activator/net/revision_backends.go

```diff
@@ -340,15 +344,47 @@ func NewRevisionBackendsManagerWithProbeFrequency(ctx context.Context, tr http.R
 			reconciler.LabelFilterFunc(networking.ServiceTypeKey, string(networking.ServiceTypePrivate), false),
 		),
 		Handler: cache.ResourceEventHandlerFuncs{
-			AddFunc:    rbm.endpointsUpdated,
-			UpdateFunc: controller.PassNew(rbm.endpointsUpdated),
+			AddFunc:    rbm.workQueue.Add,
+			UpdateFunc: controller.PassNew(rbm.workQueue.Add),
 			DeleteFunc: rbm.endpointsDeleted,
 		},
 	})
 
 	return rbm
 }
 
+func (rbm *RevisionBackendsManager) next() bool {
+	o, done := rbm.workQueue.Get()
+	if done {
+		return false
+	}
+	rbm.endpointsUpdated(o)
+	return true
+}
+
+const numThreads = 2
+
+// Run starts the backend processing goroutines in the RBM.
+func (rbm *RevisionBackendsManager) Run() {
+	var wg errgroup.Group
+
+	for i := 0; i < numThreads; i++ {
+		wg.Go(func() error {
+			for rbm.next() {
+			}
+			return nil
+		})
+	}
+	<-rbm.doneCh
+	// Make the queue to stop accepting work.
+	rbm.workQueue.ShutDown()
+	// And wait for it to drain.
+	for rbm.workQueue.Len() > 0 {
```

Comment:
Same question as in the other PR. I wonder what the semantic difference is from this loop vs. the `next` loop above 🤔

Reply:
Yeah perhaps we don't need this. I did this for uniformity.